rosetta/validator/block.go

```diff
@@ -80,9 +88,12 @@ func (v *Validator) Block(block identifier.Block) (identifier.Block, error) {
 	}
 	if block.Hash != "" && block.Hash != header.ID().String() {
 		return identifier.Block{}, failure.InvalidBlock{
-			Index:   block.Index,
-			Hash:    block.Hash,
-			Message: fmt.Sprintf("block hash does not match known hash for height (known: %s)", header.ID().String()),
+			Message: "block hash does not match known hash for height",
+			Params: map[string]interface{}{
+				"index":      block.Index,
+				"hash":       block.Hash,
```

Comment:
To illustrate my previous example, if in a future update `block.Hash` changes type and becomes a complex structure for some reason, the output of this error will change, and we will have no clue. This is very dangerous IMO and the reason why instead of passing a map of randomly typed values, each given thing should be passed to a func that expects its exact type.